xarray/core/indexes.py

```diff
@@ -702,7 +718,9 @@ def create_variables(
             encoding = None
 
         data = PandasIndexingAdapter(self.index, dtype=self.coord_dtype)
-        var = IndexVariable(self.dim, data, attrs=attrs, encoding=encoding)
+        var = IndexVariable(
+            self.dim, data, attrs=attrs, encoding=encoding, fastpath=fastpath
+        )
```

Comment:
```suggestion
        var = IndexVariable(
            self.dim, data, attrs=attrs, encoding=encoding, fastpath=True
        )
```

I think we can get away with this since we know that `data` is already wrapped and can be stuck in an `IndexVariable`. I bet we can also avoid the `isinstance(..., PandasIndexingAdapter)` check in `IndexVariable.__init__` when `fastpath is True`.

Reply:
done. lets try.